pages/api/wallets/[viewWalletPubkey].ts

```diff
@@ -1,40 +1,100 @@
 // Next.js API route support: https://nextjs.org/docs/api-routes/introduction
 import type { NextApiRequest, NextApiResponse } from 'next'
 import { Membership, PrismaClient } from '@prisma/client';
-import { MembershipModel } from '@glasseaters/hydra-sdk';
+import { Connection, clusterApiUrl, Cluster, PublicKey } from '@solana/web3.js'
 
-const prisma=new PrismaClient();
+import { useAnchorWallet, useConnection } from '@solana/wallet-adapter-react'
+import { MembershipModel, FanoutClient, Fanout } from '@glasseaters/hydra-sdk';
 
+const prisma = new PrismaClient();
+
+function validateWallet(cluster, viewWalletPubkey) {
+    const connection = new Connection(clusterApiUrl(cluster as Cluster), 'confirmed')
+    try {
+        let publickey = new PublicKey(viewWalletPubkey);
+        let a = Fanout.fromAccountAddress(connection, publickey);
```

Comment:
To handle Promise rejection in a `catch` block you should use `await`:

```typescript
let a = await Fanout.fromAccountAddress(connection, publickey);
```

For `await` to work, you would make your function `validateWallet` async:

```typescript
async function validateWallet(cluster, viewWalletPubkey) {
```

Later on, you would use `await` to handle Promise resolution.

Reply:
Ok I will do that, thanks for the input